src/edu/ufp/aed2/project/FileManager.java

```diff
@@ -0,0 +1,142 @@
+            Professor professor = new Professor(profId,profName,profCourse);
+            JsonArray studentsJsonObject = uniClass.getAsJsonArray("students");
+            ArrayList<Student> students = getStudentsFromJson(studentsJsonObject);  //students
+            Class newClass = new Class(course,type,initials,schedule,university,subject,professor);
+            classes.put(newClass,students);
+        }
+        return classes;
+    }
+
+    /**
+     * @param studentsJsonObject JsonObject from the json file
+     * @return ArrayList<Student> from the json
+     */
+    private ArrayList<Student> getStudentsFromJson(JsonArray studentsJsonObject) {
+        ArrayList<Student> students = new ArrayList<>();
+        for(JsonElement student : studentsJsonObject){
+            // iterate each student
+            JsonObject studentObject = student.getAsJsonObject();
+            String id = studentObject.get("id").toString();
+            String name = studentObject.get("name").toString();
+            students.add(new Student(id,name));
+        }
+        return students;
+    }
+
+    /**
+     * @param schedule JsonObject from the json file
+     * @return Schedule
+     */
+    private Schedule getScheduleFromJson(JsonObject schedule){
+        // getting the start InstantTime
+        JsonObject startInstanttime = schedule.getAsJsonObject("start");
+        String startDayOfWeekString = startInstanttime.get("dayOfWeek").getAsString();
+        DayOfWeek startDayOfWeek = DayOfWeek.valueOf(startDayOfWeekString);
+        String time = startInstanttime.get("time").getAsString();
+        // parse 15h30 to LocalTime(15,30)
+        LocalTime localTime = LocalTime.of(Integer.parseInt(time.substring(0,2)),Integer.parseInt(time.substring(3,5)));
+        InstantTime start = new InstantTime(startDayOfWeek,localTime);
+        // getting the end InstantTime
+        JsonObject endInstanttime = schedule.getAsJsonObject("end");
+        String endDayOfWeekString = endInstanttime.get("dayOfWeek").getAsString();
+        DayOfWeek endDayOfWeek = DayOfWeek.valueOf(endDayOfWeekString);
+        String endTime = startInstanttime.get("time").getAsString();
+        // parse 15h30 to LocalTime(15,30)
+        LocalTime endLocalTime = LocalTime.of(Integer.parseInt(time.substring(0,2)),Integer.parseInt(time.substring(3,5)));
+        InstantTime end = new InstantTime(endDayOfWeek,endLocalTime);
+        // getting the room
+        JsonObject room = schedule.getAsJsonObject("room");
+        String building = room.get("building").getAsString();
+        int number = room.get("number").getAsInt();
+        int maxSize = room.get("maxSize").getAsInt();
+        int floor = room.get("floor").getAsInt();
+        int plugNumber = room.get("plugNumber").getAsInt();
+        Room newRoom = new Room(number,building,maxSize,floor,plugNumber);
+        return new Schedule(start,end,newRoom);
```

Comment:
This function is rather extensive and complex. Perhaps it would be better to split into separate functions.